Setting a DataFeed fails downloading with NullPointer exception

I'm trying to use a datafeed URL, to use a mirror of the issues:

```scala
dependencyCheckNvdApi := NvdApiSettings(dataFeed = NvdApiSettings.DataFeed(Some(new java.net.URL("https://dependency-check.github.io/DependencyCheck_Builder/nvd_cache/"))))
```

As soon as I do that, the dependency check fails with a NullPointer exception:

```
[error] org.owasp.dependencycheck.data.update.exception.UpdateException: The execution of the download was interrupted
[error]         at org.owasp.dependencycheck.data.update.NvdApiDataSource.processDownload(NvdApiDataSource.java:283)
[error]         at org.owasp.dependencycheck.data.update.NvdApiDataSource.processDatafeed(NvdApiDataSource.java:172)
[error]         at org.owasp.dependencycheck.data.update.NvdApiDataSource.update(NvdApiDataSource.java:115)
[error]         at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:903)
[error]         at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:708)
[error]         at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:634)
[error]         at net.nmoncho.sbt.dependencycheck.tasks.package$.analyzeProject(package.scala:109)
[error]         at net.nmoncho.sbt.dependencycheck.tasks.Check$.$anonfun$apply$7(Check.scala:49)
[error]         at net.nmoncho.sbt.dependencycheck.tasks.Check$.$anonfun$apply$7$adapted(Check.scala:48)
[error]         at net.nmoncho.sbt.dependencycheck.tasks.package$.withEngine(package.scala:59)
[error]         at net.nmoncho.sbt.dependencycheck.tasks.Check$.$anonfun$apply$2(Check.scala:48)
[error]         at net.nmoncho.sbt.dependencycheck.tasks.Check$.$anonfun$apply$2$adapted(Check.scala:37)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error]         at java.base/java.lang.Thread.run(Thread.java:840)
[error] Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "org.owasp.dependencycheck.utils.Settings.getString(String)" because "this.settings" is null
[error]         at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
[error]         at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
[error]         at org.owasp.dependencycheck.data.update.NvdApiDataSource.processDownload(NvdApiDataSource.java:273)
[error]         ... 26 more
[error] Caused by: java.lang.NullPointerException: Cannot invoke "org.owasp.dependencycheck.utils.Settings.getString(String)" because "this.settings" is null
[error]         at org.owasp.dependencycheck.utils.Downloader.fetchFile(Downloader.java:446)
[error]         at org.owasp.dependencycheck.data.update.nvd.api.DownloadTask.call(DownloadTask.java:88)
[error]         at org.owasp.dependencycheck.data.update.nvd.api.DownloadTask.call(DownloadTask.java:39)
[error]         ... 4 more
```

From my look into it, it seems the 'settings' initialization for the Engine doesn't initialize the 'org.owasp.dependencycheck.utils.Downloader':

- The SBT plugins applies its settings to the `org.owasp.dependencycheck.Engine`
- The Engine later calls into `Downloader.getInstance().fetchFile()` via a `DownloadTask`
- That Downloader is a singleton with the settings in Downloader.settings. The only place that initializes that is Downloader.configure(). 
- The Download.configure() is never called? Not by the Engine nor the sbt-dependency check.

--- a/src/main/scala/net/nmoncho/sbt/dependencycheck/tasks/package.scala
+++ b/src/main/scala/net/nmoncho/sbt/dependencycheck/tasks/package.scala
@@ -39,6 +39,7 @@ import org.owasp.dependencycheck.dependency.naming.GenericIdentifier
 import org.owasp.dependencycheck.dependency.naming.Identifier
 import org.owasp.dependencycheck.dependency.naming.PurlIdentifier
 import org.owasp.dependencycheck.reporting.ReportGenerator.Format
+import org.owasp.dependencycheck.utils.Downloader
 import org.owasp.dependencycheck.utils.Settings
 import org.owasp.dependencycheck.utils.Settings.KEYS.APPLICATION_NAME
 import org.owasp.dependencycheck.utils.SeverityUtil
@@ -59,6 +60,7 @@ package object tasks {
       Thread.currentThread().setContextClassLoader(newClassLoader)
 
       try {
+        Downloader.getInstance().configure(settings)
         fn(engine)
       } catch {
         case NonFatal(e) =>

